### ISSUE TITLE
The value for the template parameter '_artifactsLocation' is not provided. 

### ISSUE THREAD

**Issue by sabbadino** [Microsoft/vsts-tasks#4490]
I have an ARM template project : Deploy-AzureResourceGroup.ps1, azuredeploy.json & azuredeploy.parameters.json 

It works fine when I deploy from VStudio & works fine in TFS using a Azure PowerShell task 
Scripth Path : NotificationEngineArmTemplate/Deploy-AzureResourceGroup.ps1
Script Arguments : -ResourceGroupName 'AzureMessageBroker-Stg' -ResourceGroupLocation 'westeurope' -TemplateFile 'azuredeploy.json' -TemplateParametersFile 'azuredeploy.parameters.json' -ArtifactStagingDirectory '.' -DSCSourceFolder '.\DSC' -UploadArtifacts

However if try to use "Azure Resource Group Deployment" 
I get "Deployment template validation failed: 'The value for the template parameter '_artifactsLocation' at line '1' and column '736' is not provided. Please see https://aka.ms/arm-deploy/#parameter-file for usage details.'."

What am I missing ?

**Comment by chrisrpatterson** [comment]
You are likely are not providing a value for that parameter in your ARM template.  If you look at the command line you passed you are probably passing the artifact location in the -ArtifactStagingDirectory parameter to the script.

**Comment by ds-ms** [comment]
@sabbadino VS let's you auto fill that field; but you'll have to manage it yourself when you're doing it by yourself.

**Comment by Ranu2046** [comment]
I too have similar kind of issue.. when i deploy the ARM template from Visual studio then all the parameters are auto populated from parameter file but when i try to deploy running the  .PS1 (powershell script), It prompts to enter the values for all the parameters present in the parameter file. Can anyone please let me know how to auto read the parameter file without prompting user to enter the same,

**Comment by sachinma** [comment]
@Ranu2046 You can use the Azure Resource Group Deployment task to perform this operation which takes both a template file and a parameters file as input.

**Comment by jfoshee** [comment]
I completely agree with the OP in that this is confusing.  I'd like to see Microsoft add a better VSTS task to support Azure Resource Group projects.  

In the mean time, let me correct a misconception in this thread: It is not Visual Studio that auto-populates `_artifactsLocation` and `_artifactsLocationSasToken`. Rather it is the PowerShell script that runs as part of the deployment; in the project it is probably called `Deploy-AzureResourceGroup.ps1`.  In that script you can see the logic for populating those parameters.

The best thing you can do is read [this excellent article which details two ways to deploy Azure Resource Group Projects in a VSTS pipeline](https://docs.microsoft.com/en-us/azure/vs-azure-tools-resource-groups-ci-in-vsts).

To summarize, you are probably better off using an **Azure PowerShell Task** to run `Deploy-AzureResourceGroup.ps1`. I know it is less than ideal, but in this way you are doing the same thing locally as in your CD pipeline.
